core/domain/user_services.py

```diff
@@ -933,22 +976,262 @@ def create_new_user(gae_id, email):
+        user_settings = UserSettings(
+            user_id, gae_id, email, feconf.ROLE_ID_LEARNER,
+            preferred_language_codes=[constants.DEFAULT_LANGUAGE_CODE]
+        )
+        if (not modifiable_user_data.display_alias or
+                not isinstance(
+                    modifiable_user_data.display_alias,
+                    python_utils.BASESTRING
+                )
+           ):
+            raise utils.ValidationError(
+                'Expected display alias to be a string, received %s' %
+                modifiable_user_data.display_alias
+            )
+        user_settings.display_alias = modifiable_user_data.display_alias
+        user_settings.last_agreed_to_terms = (
+            modifiable_user_data.last_agreed_to_terms)
+        user_settings.last_logged_in = modifiable_user_data.last_logged_in
+        user_settings.user_bio = modifiable_user_data.user_bio
+        user_settings.subject_interests = (
+            modifiable_user_data.subject_interests)
+        user_settings.preferred_language_codes = (
+            modifiable_user_data.preferred_language_codes)
+        user_settings.preferred_site_language_code = (
+            modifiable_user_data.preferred_site_language_code)
+        user_settings.preferred_audio_language_code = (
+            modifiable_user_data.preferred_audio_language_code)
+
+        user_auth_details = UserAuthDetails(
+            user_id, None, modifiable_user_data.pin, parent_user_id)
+
+        transaction_services.run_in_transaction(
+            _create_new_profile_transactional, user_settings,
+            user_auth_details
+        )
+        user_settings_list.append(user_settings)
+    return user_settings_list
+
+
+def update_multiple_users_data(modifiable_user_data_list):
+    """Updates user settings and user auth model details for the users
+    specified in the modifiable_user_data_list.
+
+    Args: user_details_change_list: list(ModifiableUserData). The list of
+            modifiable_user_data entries corresponding to the users whose
+            data has to be updated.
+
+    Raises:
+        ValidationError. If a None or empty value is provided for display
+            alias attribute.
+        Exception. If a None or empty value is provided for user_id.
+    """
+    user_settings_list = []
+    user_auth_details_list = []
+    for modifiable_user_data in modifiable_user_data_list:
+        user_id = modifiable_user_data.user_id
+        if not user_id:
+            raise Exception(
+                'A valid user id must exist for all the users to be updated.')
+        user_settings = get_user_settings(user_id, strict=True)
+        if not modifiable_user_data.display_alias:
+            raise utils.ValidationError(
+                'Expected display_alias to be a string, received %s.'
+                % (modifiable_user_data.display_alias)
+            )
+        user_settings.display_alias = modifiable_user_data.display_alias
```

Comment:
You should have something like user_settings.populate_from_profile_data(...)

Reply:
Done.